Resources/public/js/apps/ez-platformuiapp.js

```diff
@@ -140,8 +142,20 @@ YUI.add('ez-platformuiapp', function (Y) {
 
             this.on({
                 'loadingChange': this._loading,
-                'navigate': function () {
+                '*:contentDiscover': function() { this.set('isDiscovering', true)},
```

Comment:
actually, after thinking about it, maybe you can try something:
in eZ.Plugin.UniversalDiscovery when showing the universal discovery you could manipulate the application's routes (either completely remove them or just changing the middlewares for a *no operation* middleware) so that the navigate becomes impossible. And of course, when hiding the UDW, you should restore the app that the ability to navigate is restored.